### PR TITLE
Implemented changes to resolve #89

### LINKED ISSUE #89: Problems using string methods as condition
### DESCRIPTION
**Describe the bug**
Using a stirng method for a `if` condition yields to problems, especially using `if (strvar.fnd(","))`. The problem might be related to the comma in the string expression.

**To Reproduce**
1. Reproduce a code like in the screenshot
2. Execute the code
3. See error

**Expected behavior**
The code is executed without problems

**Screenshots**
grafik

**Desktop (please complete the following information):**
 - OS: 10x64
 - Version v1.1.5.2303

**Additional context**

(*Do not write below this line*)

-----------------------------------------------------
### ANALYSIS
The Problem originates from a missing condition in `value_type* FlowCtrl::evalHeader(int& nNum, std::string& sHeadExpression, bool bIsForHead, int nth_Cmd, const std::string& sHeadCommand)`, where the `CALCTYPE_NUMERICAL` is assigned always even if other previous evaluations are necessary (compare that with `int FlowCtrl::compile(std::string sLine, int nthCmd)`)

### IMPLEMENTATION STEPS
(*see also our [Wiki for implementation guidelines](https://github.com/numeredev/NumeRe/wiki/HowTo:-Contribute-Code)*)
- Implement the necessary changes in a new branch created here on GitHub
- Test your implementation

### DOCUMENTATION STEPS
(*see also our [Wiki for further information](https://github.com/numeredev/NumeRe/wiki/HowTo:-Create-Language-Strings)*)
- Update the changes log
- Add comments to your implementation
- Add Doxygen documentation comments
- Create or update the documentation articles (`*.NHLP` and `*.NDB` files, if needed)
- Update the language strings (`*.NLNG` files, if needed)

### PULL REQUEST
- Create a pull request for your changes
- Fill out the template
- Assign @numere-org/maintainers as reviewers

### PR DESCRIPTION
**Pull request by numeredev**
### SUMMARY
Implements necessary changes for #89 

*Reviewers:* @numere-org/maintainers

### IMPLEMENTATION
* Implementation: The problem was located in a completely different location. The issue was the comma used as part of the string literal within the string method. This could not be resolved correctly. Changed the logic now.
* Implementation test: Re-executed the code without problems.

### DOCUMENTATION
* [x] ChangesLog updated
* [x] Code changes commented
* **Documentation articles:**
    * [ ] corresponding documentation articles updated
    * [ ] new documentation articles created
    * [x] not needed
* **Language files:**
    * [ ] corresponding language files updated
    * [x] not needed

----------------

### TESTS BY REVIEWERS
* [x] Added to the automatic SW tests
* [x] Tested manually

### CODE REVIEW

**Comment by numeredev** [comment]
Use something like this to simply replicate the error:

```
mystring = "12345";
mystring.fnd(",")
```

**Comment by numeredev** [comment]
Sneaked in a small change to resolve the glitch that the necessary byte size is calculated to small for string based columns.

File: kernel/core/datamanagement/memory.cpp [hunk 1]
```diff
@@ -841,7 +841,7 @@ void Memory::copyElementsInto(vector<mu::value_type>* vTarget, const VectorIndex
 bool Memory::isValidElement(size_t _nLine, size_t _nCol) const
 {
     if (_nCol < memArray.size() && _nCol >= 0 && memArray[_nCol])
-        return !mu::isnan(memArray[_nCol]->getValue(_nLine));
+        return memArray[_nCol]->isValid(_nLine); // THIS CHANGE MIGHT HAVE BROKEN OTHER THINGS
```
Comment (raphael-sys) [inline kernel/core/datamanagement/memory.cpp:844]: Is this still relevant? Should this be removed or a TODO?

**Review by raphael-sys — APPROVED** [review]

File: kernel/core/datamanagement/memory.cpp (hunk 1 above)
Comment (numeredev) [reply kernel/core/datamanagement/memory.cpp:844]: Still relevant. Might yield problems in final tests and is therefore easier to find

**Pull request merged**